soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+
+#include <am_mcu_apollo.h>
+
+static int arm_apollo4_init(void)
+{
+	am_hal_pwrctrl_low_power_init();
+	am_hal_pwrctrl_control(AM_HAL_PWRCTRL_CONTROL_SIMOBUCK_INIT, 0);
```

Comment:
if controlling a buck converter, this likely needs a regulator driver. could you provide some reference/guiding here?

Reply:
It turned out that the default configuration is sufficient for basic operations. We'll contribute a proper regulator driver in a separate PR.